src/app/modules/submission/submission.middleware.ts

```diff
@@ -0,0 +1,49 @@
+import { RequestHandler } from 'express'
+import { ParamsDictionary } from 'express-serve-static-core'
```

Comment:
could you remind me again why it is necessary that we have a dependency on the [express-serve-static-core](https://www.npmjs.com/package/express-serve-static-core) package which has less than 8000 downloads/week? It's usage appear to be quite prevalent in our controllers.

Reply:
This is for typing, express itself does not export `ParamsDictionary`

Reply:
`Express.RequestHandler` has the following interface:
```
interface RequestHandler<P = core.ParamsDictionary, ResBody = any, ReqBody = any, ReqQuery = core.Query> extends core.RequestHandler<P, ResBody, ReqBody, ReqQuery> { }
```

`ParamsDictionary` is the default for the first generic type, so we often just insert it so we can subsequently specify `ResBody` and `ReqBody`. but in hindsight, using `unknown` would have been a better idea, since `ParamsDictionary` is of type `{ [key: string]: string; }`, which may lead one to wrongly assume that a parameter is defined.

`express-serve-static-core` exists just to make the types for `express` work, which is why it has so few downloads.